### ISSUE TITLE
Table panel ignore cell display mode / thresholds set in overrides 

### ISSUE THREAD

**Issue by cykl** [grafana/grafana#43477]
**What happened**:

Within a table panel with cell display mode set to _auto_, I tried to override one field to have cell display mode set to _color background (solid)_ and defined some thresholds in the override (numeric field). 

I expected  the background of this column to have a solid color based on cell value and threshold. However, I observed no change and cells remained with a transparent background. 

**What you expected to happen**:

The override should have be applied

**How to reproduce it (as minimally and precisely as possible)**:

Use an Elasticsearch data source. 

Create a table panel with a "Raw Data" metrics

Configure the table to have cell display mode set to Auto (default)

![image](https://user-images.githubusercontent.com/1764725/147132876-54ccb559-644a-41ed-896d-fe81d58ac52f.png)

Add overrides on some numeric columns. Set cell display mode to color background solid and define some thresholds.

![image](https://user-images.githubusercontent.com/1764725/147133098-2fc6c2bf-d050-48f1-9b8b-9da9f3cf31ce.png)

Observe overrides do not work has expected

![image](https://user-images.githubusercontent.com/1764725/147133750-20bf2b26-3531-43a0-bae8-92176f26efe1.png)

**Anything else we need to know?**:

* Default "cell display mode" doesn't seem to matter
* Setting "cell display mode" to LCD in override actually change the cell display, however colors do not reflect the thresholds

**Environment**:
- Grafana version: 8.3.3
- Data source type & version: Elasticsearch
- OS Grafana is installed on:
- User OS & Browser: Firefox 95, Linux

**Comment by zuchka** [comment]
Thanks for creating this issue, @cykl 

Have you checked that the status codes are numbers and not strings? Try using the [`Change field type` transformation](https://grafana.com/docs/grafana/latest/panels/transformations/types-options/#convert-field-type).

Otherwise, I would advise uploading your raw, unformatted data [using these steps](https://cln.sh/PTgRFb). This way the squads can try and mock your panel and data.

**Comment by grafanabot** [comment]
Thanks for creating this issue! We think it's missing some basic information. 

Follow the issue template and add additional information that will help us replicate the problem. 
For data visualization issues: 
- Query results from the inspect drawer (data tab & query inspector)
- Panel settings can be extracted in the panel inspect drawer JSON tab

For dashboard related issues: 
- Dashboard JSON can be found in the dashboard settings JSON model view

For authentication, provisioning and alerting issues, Grafana server logs are useful. 

Happy graphing!

**Comment by grafanabot** [comment]
Thanks for creating this issue! We think it's missing some basic information. 

Follow the issue template and add additional information that will help us replicate the problem. 
For data visualization issues: 
- Query results from the inspect drawer (data tab & query inspector)
- Panel settings can be extracted in the panel inspect drawer JSON tab

For dashboard related issues: 
- Dashboard JSON can be found in the dashboard settings JSON model view

For authentication, provisioning and alerting issues, Grafana server logs are useful. 

Happy graphing!

**Comment by cykl** [comment]
@zuchka 

I was assuming that column type would be numeric because those fields are int / float in Opensearch but you might be on the right track. 

Query inspector's data tab shows the column type as string

![image](https://user-images.githubusercontent.com/1764725/147201762-7be2e3ef-9dda-4f45-a59e-7df3fcb6862f.png)

However, if I disable "Formatted data" they are show as number

![image](https://user-images.githubusercontent.com/1764725/147201985-ca166cd3-1a1f-4823-929d-96154a255485.png)

Inspecting the raw response confirms that those fields are JSON numbers

```json
0:Object
  _index:"redacted"
  _type:"_doc"
  _id:"redacted"
  _score:null
  _source:Object
    response_time:3.7900888919830322
    status_code:201
    @timestamp:"2021-12-23T07:06:41.994982Z"
```

What is strange is that gauges kind of work (colors don't work but higher number -> longer bar). 

Here are my query result & panel settings slightly edited to redact fields that I cannot share. If that not good enough, I will try to create a repoducer from scratch:

* [Access logs -data-2021-12-23 08 12 58.csv](https://github.com/grafana/grafana/files/7767415/Access.logs.-data-2021-12-23.08.12.58.csv)
* [table_panel.json.txt](https://github.com/grafana/grafana/files/7767436/table_panel.json.txt)

**Comment by zoltanbedi** [comment]
Hi @cykl,
I've tried to recreate your issue with the data provided and found that the color scheme set to single color. When I changed it to from threshold I think I got the expected behaviour.
![Screenshot 2022-01-11 at 17 02 42](https://user-images.githubusercontent.com/13729989/148978035-4c325154-9887-43b6-bce0-ddab24086087.png)

I'm going to close this issue. Let me know if you still think there is a bug we missed and we can reopen it.
